= *MemPanG23*
:figure-caption!:
:toc-title: Page Content
:toc: left
:toclevels: 2

A computational pangenomics course and virtual conference in Memphis, TN.

image::images/DRB1-3123.fa.gz.pggb-E-s5000-l15000-p80-n10-a0-K16-k8-w50000-j5000-e5000-I0-R0-N.smooth.chop.og.lay.draw_mqc.CROP.png[400]
image::images/bridge.png[400]


=== *Important Dates*

- Practical Course Registration Deadline: *15th April 2023*
- Selected Participant will be notified by: *20th April 2023*
- Payment Deadline: *29th May 2023*
- Virtual Conference Registration Deadline: *29th May 2023*

=== *Course Venue*

University of Tennessee Health Science Center +
Mooney Building, +
Library, Room 201, +
875 Monroe Ave., +
Memphis, TN 38163 +
United States +
-> https://goo.gl/maps/fCZbYYeGyUko353B8[See the venue on the map]


==== *Hotels in Memphis*
- link:https://goo.gl/maps/arRZWkjPCNBAFfCf9[La Quinta Hotel Memphis Downtown]
- link:https://goo.gl/maps/Dwf9LgHeJLXsAJcf7[Double Tree Hotel]
- link:https://goo.gl/maps/h5A6LpmToTD7DepH8[River Inn at Harbor Town Hotel]
- link:https://goo.gl/maps/o7XrW3DZHTNqXXT26[Comfort Inn]
- link:https://goo.gl/maps/e365A6rNjZUPvFqRA[Hampton Inn]
- link:https://goo.gl/maps/rjdSg46kZFPsmKxS7[Hyatt Centric Beale Street]
- link:https://goo.gl/maps/e4BcvEabefsqEhC4A[HU Hotel]
- link:https://goo.gl/maps/Erq5cwVtM4hh8c8e7[Sheraton Memphis Downtown Hotel]

==== *Transportation in Memphis*

If you stay in Downtown, the Medical District, South City, or New Chicago, you can reach the course venue by getting on-demand, convenient, and affordable rides with link:https://city.ridewithvia.com/groove-on-demand[Groove On-Demand].

== *About the Practical Course*

*Tuesday, May 30, 2023 - Thursday, June 1, 2023* +
The use of a single reference genome in bioinformatics can lead to reference bias and miss important information about genome variability and relationships. New assembly methods have made it easier to generate high-quality complete genome assemblies, and using a pangenome graph that expresses many genomes and their mutual alignment can help address these issues. This course will focus on building such graphs from whole genome assemblies and using them in various downstream applications in comparative genomics, evolution, variation analysis, sequence alignment, and phenotype association.

Participants will learn about pangenome concepts and gain practical experience building and analyzing pangenome graphs. They will apply these methods to complex research questions that require understanding the relationships between multiple genomes or accounting for variability when analyzing new genomes. By the end of the course, participants will have a strong understanding of pangenome methods based on whole genome assemblies.


== *About the Virtual Conference*

*Friday, June 2, 2023* +
We will gather for a hybrid in-person and virtual conference on the last day of the week.
This will bring together course participants and virtual and in-person speakers who are actively working on diverse aspects of pangenome research.
The virtual meeting will be open to the public.
Speakers will cover topics at the intersection of pangenome methods and biology, including a focus on basic methods, data structures, genome assembly, data visualization, and public collaborative pangenome projects.
We expect to draw speakers from a diverse set of species contexts.

== *People*

=== *Practical Course Instructors*

==== Erik Garrison - Assistant Professor

.[purple]#University of Tennessee Health Science Center, Memphis, TN, US#

image:images/erik.jpeg[erik,200,role="right"] Genomicist with a quantitative social science background. I build methods that let us understand the precise relationships between thousands of genomes. In these, the genome is encoded in a graph that may represent a population sample of individuals from the same species, a metagenome, the diploid genome of a single individual, or any other useful collection of sequences.

image:images/Octicons-mark-github.svg[git,30] https://github.com/ekg[GitHub]

'''

==== Andrea Guarracino - Postdoctoral Scholar

.[purple]#University of Tennessee Health Science Center, Memphis, TN, US#

image:images/andrea.jpeg[erik,200,role="right"]Computer (bio)scientist dedicated to developing and applying new methods to analyze ever-larger collections of sequences (also called "pangenomes") in the tree of life.

image:images/Octicons-mark-github.svg[git,30] https://github.com/andreaguarracino[GitHub]

'''

==== Simon Heumos - PhD student

.[purple]#Quantitative Biology Center (QBiC), University of Tübingen, Tübingen, DE | Biomedical Data Science, Department of Computer Science, University of Tübingen, Tübingen, DE#

image:images/simon.png[simon,200,role="right"]

Bioinformatician by training. I am exploring methods to calculate and visualize pangenome graph layouts. These are key steps in pangenome graph construction and analysis pipelines. Cluster-scalable pangenomics is the next step.

image:images/Octicons-mark-github.svg[git,30] https://github.com/subwaystation[GitHub]

'''

==== Arang Rhie - Staff Scientist

.[purple]#Genome Informatics Section, Computational and Statistical Genomics Branch, National Human Genome Research Institute, National Institutes of Health, Bethesda, MD, USA#

image:images/arang.jpeg[arang,200,role="right"]

I am an active member of the Vertebrate Genomes Project (VGP), Earth Biogenome Project (EBP), Human Pangenome Reference Consortium (HPRC) and the Telomere-to-Telomere consortium (T2T).

image:images/Octicons-mark-github.svg[git,30] https://github.com/arangrhie[GitHub]

'''

==== Jordan Eizenga - Postdoctoral Researcher

.[purple]#UC Santa Cruz Genomics Institute, University of California, Santa Cruz, Santa Cruz, CA, USA#

image:images/jordan.webp[jordan,200,role="right"]

My research focuses on a variety of topics in and around computational pangenomics. I am part of VG team.

image:images/Octicons-mark-github.svg[git,30] https://github.com/jeizenga[GitHub]

'''

=== *Practical Course Helpers*


==== Flavia Villani - PhD student

.[purple]#University of Tennessee Health Science Center, Memphis, TN, US#
image:images/Flavia.JPG[flavia,200,role="right"] I have a master degree in Medical Biotechnology from the University of Naples Federico II. Currently, I am a PhD student at the University of Tennessee Health Science Center. I am building the pangenome of model organisms (inbred mice and rats) using a combination of second and third generation sequence data.

image:images/Octicons-mark-github.svg[git,30] https://github.com/Flavia95[GitHub]


'''
==== Christian Fischer - IT Analyst III

.[purple]#University of Tennessee Health Science Center, Memphis, TN, US#
image:images/christian.jpeg[christian,200,role="right"]Computer scientist also trained in interaction design, with a love for mathematics. I work on tooling for variation graphs using the Rust programming language, with a special interest in visualization, which is currently channeled into my pangenome visualization tool Waragraph.


image:images/Octicons-mark-github.svg[git,30] https://github.com/chfi[GitHub]

{empty} +


=== *Virtual Conference Invited speakers*

- link:https://scholar.google.com/citations?user=jfexpLUAAAAJ&hl=en[Zhigui Bao]
- link:https://scholar.google.com/citations?hl=en&user=XzUOb7IAAAAJ[Arang Rhie]
- link:https://scholar.google.com/citations?user=t0Sk55AAAAAJ&hl=en&oi=ao[Scott Edwards]
- link:https://scholar.google.com/citations?hl=en&user=wpPBcf4AAAAJ&view_op=list_works&sortby=pubdate[Christina Boucher]
- link:https://scholar.google.com/citations?user=z2edTLkAAAAJ&hl=en[Joep de Ligt]
- link:https://scholar.google.com/citations?user=yp4X1F8AAAAJ&hl=en&oi=ao[Peter Sudmant]

=== *Organizers*

- link:https://andreaguarracino.github.io/[Andrea Guarracino], University of Tennessee Health Science Center, Memphis, TN, US
- link:http://hypervolu.me/~erik/erik_garrison.html[Erik Garrison], University of Tennessee Health Science Center, Memphis, TN, US
- link:https://github.com/Flavia95[Flavia Villani], University of Tennessee Health Science Center, Memphis, TN, US
- link:https://scholar.google.com/citations?user=rWNHPpAAAAAJ&hl=en[Jordan M Eizenga], University of California Genomics Institute, Santa Cruz, CA, US
//- link:https://uthsc.edu/cbmi/big/big-faculty.php[Melanie Hayes], University of Tennessee Health Science Center, Memphis,TN,US
- link:https://thebird.nl/[Pjotr Prins], University of Tennessee Health Science Center, Memphis, TN, US
- link:https://www.uthsc.edu/faculty/profile/?netid=rdavis88[Robert Davis], University of Tennessee Health Science Center, Memphis, TN, US
- link:https://www.uthsc.edu/faculty/profile/?netid=rwilli10[Robert Williams], University of Tennessee Health Science Center, Memphis, TN, US
- link:https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/team/[Simon Heumos], University of Tuebingen Quantitative Biology Center, Tuebingen, DE
//- link:https://www.uthsc.edu/genetics/faculty-staff.php[Tamara Brock], University of Tennessee Health Science Center, Memphis,TN,US
- link:https://www.uthsc.edu/faculty/profile/?netid=vcolonna[Vincenza Colonna], University of Tennessee Health Science Center, Memphis, TN, US


image:images/UTHSC.png[uthsc,300]

University of Tennessee Health Science Center, Department of Genetics, Genomics and Informatics

College of Medicine, Department of Pediatrics

== *Program*

=== *Practical Course (Central Time Zone)*

*Zoom link*:https://tennesseehipaa.zoom.us/j/96698730457?pwd=TXNEbzRaVWRkNVhJY3dBalZGdGVzQT09

*Place*: link:https://goo.gl/maps/tLySYa8zdLsUhE2C6[875 Monroe Ave, Mooney Building, Nash Atrium]

[options="header", cols="2,1,2,4"]
|===
|Day | Time | Speaker(s) | Topic

|Tuesday, May 30, 2023 | 14:30-16:00 | link:http://hypervolu.me/~erik/erik_garrison.html[Erik Garrison] | Presentation: link:https://docs.google.com/presentation/d/1tkWQ6I-oD73MUPGzw3pc1waJ1w2_sx528K188w1srTM/edit?usp=sharing[Introduction to pangenomics]
|| 16:00-16:30 | Everyone | Coffee break
|| 16:30-17:15 | Everyone | Practical: link:https://hackmd.io/@AndreaGuarracino/HkrROK_V3[HLA & LPA graphs, PGGB]
|| 17:15-17:30 | Everyone | Q&A, Day 1 Survey

|Wednesday, May 31, 2023 | 09:30-10:00 | link:https://andreaguarracino.github.io/[Andrea Guarracino] | Presentation: link:https://docs.google.com/presentation/d/1IXs7HDdpCZBY-MapoGaZe0P-iVwgZfgpApC0vQ5icJ0/edit?usp=sharing[Understanding pangenomes]
|| 10:00-10:30 | Everyone | Practical: link:https://hackmd.io/@AndreaGuarracino/SyhbiKuE2[Human pangenome graph building]
|| 10:30-11:00 | Everyone | Coffee break
|| 11:00-12:30 | Everyone | Practical: link:https://hackmd.io/@AndreaGuarracino/SyhbiKuE2[ODGI and community detection]
|| 12:30-14:30 | Everyone | Lunch

|| 14:30-15:00 | link:https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/team/[Simon Heumos] | Presentation: nf-co.re/pangenome, pangenome openness
|| 15:00-16:00 | Everyone | Practical: nf-co.re/pangenome, pangenome openness
|| 16:00-16:30 | Everyone | Coffee break
|| 16:30-17:15 | Everyone | Practical: nf-co.re/pangenome, pangenome openness
|| 17:15-17:30 | Everyone | Q&A, Day 2 Survey

|Thursday, June 1, 2023 | 09:30-10:00 | link:https://scholar.google.com/citations?user=rWNHPpAAAAAJ&hl=en[Jordan M Eizenga] | Presentation: Read mapping with vg giraffe
|| 10:00-10:30 | Everyone | Practical: alignment and variant calling
|| 10:30-11:00 | Everyone | Coffee break
|| 11:00-12:30 | Everyone | Practical: alignment and variant calling
|| 12:30-14:30 | Everyone | Lunch

|| 14:30-15:00 | link:https://genomeinformatics.github.io/people/rhie/[Arang Rhie] | Presentation: State of the art in genome assembly
|| 15:00-16:00 | Everyone | Practical: T2T assembly problems
|| 16:00-16:30 | Everyone | Coffee break
|| 16:30-17:15 | Everyone | Practical: T2T assembly problems
|| 17:15-17:30 | Everyone | Q&A, Day 3 Survey
|===

=== *Virtual Conference (Central Time Zone)*

*Time:* Friday, June 2, 2023 +
*Place:* link:https://goo.gl/maps/tLySYa8zdLsUhE2C6[875 Monroe Ave, Mooney Building, Mooney Library] +
*Zoom link*: https://tennesseehipaa.zoom.us/j/92881122014

[options="header", cols="2,2,5,2"]
|===
|Chair | Time | Talk | Speaker

|| 09:20 - 9:30 | Conference Opening | link:https://scholar.google.com/citations?user=OYJMYwIAAAAJ&hl=en[Rob Williams]

|link:https://andreaguarracino.github.io/[Andrea Guarracino] | 09:30 - 10:00 | Hidden complex genome rearrangements in the plant pangenome graph | link:https://scholar.google.com/citations?user=jfexpLUAAAAJ&hl=en[Zhigui Bao]
|| 10:00 - 10:30 | Beyond one T2T human genome: What's next? | link:https://scholar.google.com/citations?hl=en&user=XzUOb7IAAAAJ[Arang Rhie]
|| 10:30 - 11:00 | Coffee break |

|link:https://uni-tuebingen.de/forschung/forschungsinfrastruktur/zentrum-fuer-quantitative-biologie-qbic/team/[Simon Heumos] | 11:00 - 11:30 | Unexpected complexity and abundant structural variation in the pangenomes of birds | link:https://scholar.google.com/citations?user=t0Sk55AAAAAJ&hl=en&oi=ao[Scott Edwards]
|| 11:30 - 12:00 | Building a Pangenome Alignment Index via Recursive Prefix-Free Parsing  | link:https://scholar.google.com/citations?hl=en&user=wpPBcf4AAAAJ&view_op=list_works&sortby=pubdate[Christina Boucher]
|| 12:00 - 14:00 | Lunch |

|link:https://github.com/Flavia95[Flavia Villani] | 14:00 - 14:30 | Pangenome graphs in pathogen genomics for public health | link:https://scholar.google.com/citations?user=z2edTLkAAAAJ&hl=en[Joep de Ligt]
|| 14:30 - 15:00 | Recent human evolution of complex genome architectures | link:https://scholar.google.com/citations?user=yp4X1F8AAAAJ&hl=en&oi=ao[Peter Sudmant]
|===


== *Registration and Practical Information*

Registration includes access to: all lectures and practical sessions, all course materials.

- link:https://forms.gle/oeku3F3crTDb5N8F6[Practical Course Registration form]

- link:https://forms.gle/PRKcQsz5Aivj1qg19[Virtual Conference Registration form]

=== *Cost*

The cost is intended for the course. The conference is free. We do not expect meals to be included.

- Academic - $200, with financial support if needed
- Industry - $400

=== *Selection criteria*

This course is intended for biologists and bioinformaticians interested in studying organisms with high genetic diversity or without a reference genome, as well as those involved in comparative genomics and the assembly of pangenomes for any species.

Selection of participants will be based on:

- good knowledge of Linux operating system and basic shell commands. This will be a mandatory prerequisite.
- familiarity with genomics data formats (e.g., FASTA, VCF, BED, ...) is a plus.
- impact of the course for the participant and his/her research group.
- stage of the research project: priority will be given to participants with data already available and ready to be analyzed (participants data will not be analyzed during the course).

Fulfillment of these conditions by participants will be assessed through the registration form.

[cols=">a,<a,<a", frame=none, grid=none]
|===
| image::images/memphis.jpg[width=300,height=300]
|image::images/men.jpg[width=600,height=400]
| image::images/bridge.jpg[width=300,height=300]
|===

https://www.memphistravel.com/sports-outdoors?gclid=Cj0KCQiAgOefBhDgARIsAMhqXA4Gt_kloBAAqe5LDWNW3199TI8DzRrUo4fCqSQ_cKLkRPd4xv46TUgaAt4WEALw_wcB[memphistravel]

{empty} +
{empty} +



== *Contact*

- mailto:genetics@uthsc.edu[genetics@uthsc.edu]
