# MEMPanG23
The computational pangenomics course in Memphis.

## What
The use of a single reference genome in bioinformatics can lead to reference bias and miss important information about genome variability and relationships. New assembly methods have made it easier to generate high-quality complete genome assemblies, and using a pangenome graph that expresses many genomes and their mutual alignment can help address these issues. This course will focus on building such graphs from whole genome assemblies and using them in various downstream applications in comparative genomics, evolution, variation analysis, sequence alignment, and phenotype association.

## For who
This course is intended for biologists and bioinformatics interested in studying organisms with high genetic diversity or without a reference genome, as well as those involved in comparative genomics and the assembly of pangenomes for any species.

## Why
Participants will learn about pangenome concepts and gain practical experience building and analyzing pangenome graphs. They will apply these methods to complex research questions that require understanding the relationships between multiple genomes or accounting for variability when analyzing new genomes. By the end of the course, participants will have a strong understanding of pangenome methods based on whole genome assemblies.

## Where
[University of Tennessee Health Science Center](https://uthsc.edu/)<br/>
XXXXXXXX building<br/>
Memphis, Tennessee<br/>
United States

## When
May 30th - Jun 2st 2023

## Who
Instructors:

- [Erik Garrison](https://github.com/ekg), University of Tennessee Health Science Center, Memphis, TN, US
- [Andrea Guarracino](https://github.com/AndreaGuarracino), University of Tennessee Health Science Center, Memphis, TN, US

Helpers:

- [Flavia Villani](https://github.com/Flavia95), University of Tennessee Health Science Center, Memphis, TN, US
- [??? ???](https://github.com/???), ???

Invited speakers:

- [AAA](https://github.com/AAA)
- [BBB](https://github.com/BBB)
- [CCC](https://github.com/CCC)
- [DDD](https://github.com/DDD)


## PROGRAM

### May 30 - COURSE - Day 1:
    - Session 1
        - Presentation: introduction (what is a pangenome, why we need it, pangenome graph, variation graph, GFAv1, taste of pangenome graph construction)
        - Practical activities: simple pangenome graphs construction
    - Session 2
        - Presentation: PanGenome Graph Builder
        - Practical activities: on HLA and LPA pangenome graphs

### May 31 - COURSE - Day 2:
    - Session 1
        - Presentation: Pangenome visualization (in 1D and 2D), graph sorting
        - Practical activities: Human pangenome graphs
    - Session 2
        - Presentation: ODGI (graph extraction and untangling)
        - Practical activities with ODGI ()

### June 1 - COURSE - Day 3:
    - Session 1
        - Presentation: ***MISSING***
        - Practical activities: Saccharomyces cerevisiae (with sequence divergence estimation)
    - Session 2
        - Presentation: ***MISSING***
        - Practical activities:
            - Saccharomyces cerevisiae (with community detection)
            - OPTIONAL: simple read mapping and variant calling (vg map/pack/call)

    ### NOTE: we could replace this part with a genotyping session (bwa-mem, gfa-inject,gfapack, cosine distance)

### June 2 - CONFERENCE 
        - 09:30 - 10:00 talk 1
        - 10:00 - 10:30 talk 2
        - 10:30 - 11:00 break 
        - 11:00 - 11:30 talk 3
        - 11:30 - 12:00 talk 4
